lib/zip.js

```diff
@@ -33,22 +60,56 @@ function addVendorHelper() {
  */
 function removeVendorHelper() {
   if (this.options.zip && this.options.cleanupZipHelper) {
-    this.serverless.cli.log('Removing Python requirements helper...');
-    return fse.removeAsync(path.join(this.servicePath, 'unzip_requirements.py'));
+    if (this.serverless.service.package.individually) {
+      let promises = [];
+      let doneModules = [];
+      values(this.serverless.service.functions)
+        .forEach((f) => {
+          if (!doneModules.includes(f.module)) {
+            this.serverless.cli.log(`Removing Python requirements helper from ${f.module}...`);
+            promises.push(
+              fse.removeAsync(path.join(this.servicePath, f.module, 'unzip_requirements.py'))
+            );
+            doneModules.push(f.module);
+          }
+        });
+      return BbPromise.all(promises);
+    } else {
+      this.serverless.cli.log('Removing Python requirements helper...');
+      return fse.removeAsync(path.join(this.servicePath, 'unzip_requirements.py'));
+    }
   }
 };
 
 /**
  * zip up .serverless/requirements
- * @return {Promise}
  */
 function packRequirements() {
   if (this.options.zip) {
-    this.serverless.cli.log('Zipping required Python packages...');
-    this.serverless.service.package.include.push('.requirements.zip');
-    return zipDirectory(
-                path.join(this.servicePath, '.serverless/requirements'),
-      path.join(this.servicePath, '.requirements.zip'));
+    if (this.serverless.service.package.individually) {
+      let doneModules = [];
+      values(this.serverless.service.functions)
+        .forEach((f) => {
+          if (!doneModules.includes(f.module)) {
+            this.serverless.cli.log(`Zipping required Python packages for ${f.module}...`);
+            f.package.include.push(`${f.module}/.requirements.zip`);
+            zipper
+              .sync
+              .zip(`.serverless/${f.module}/requirements`)
+              .compress()
+              .save(`${f.module}/.requirements.zip`);
+            doneModules.push(f.module);
```

Comment:
Nice - I like the caching on this to avoid re-packing deps for modules that have multiple entry points